code/services/QueuedJobService.php

```diff
@@ -615,14 +615,20 @@ public function runJob($jobId) {
 
 		Config::unnest();
 
-		// okay lets reset our user if we've got an original
+		// okay let's reset our user if we've got an original
 		if ($runAsUser && $originalUser) {
 			Session::clear("loggedInAs");
 			if ($originalUser) {
 				Session::set("loggedInAs", $originalUser->ID);
 			}
 		}
 
+		// let's make sure there is a cleanupJob in the queue
+		if (Config::inst()->get('CleanupJob', 'is_enabled')) {
```

Comment:
It could - I did consider that behaviour. As it is already disabled by default, I'm not sure which I prefer. I think it ultimately comes down to who you want controlling whether or not the cleanup job runs - the CMS admin, or the dev. My gut says the dev, and that it should be controlled by config, but I guess ultimately it is up to you as module maintainer. Happy to change it over to be a self-restarting job if that's what you think is best.